indexclient/client.py

```diff
@@ -320,6 +320,63 @@ def list_versions(self, did):
             versions.append(Document(self, version["did"], version))
         return versions
 
+    def query_urls_metadata(self, url, key, value, fields=None, versioned=False, limit=100, offset=0, page_size=100):
+        params = {
+            "url": url,
+            "key": key,
+            "value": value,
+            "fields": fields,
+            "versioned": versioned,
+            "limit": limit if limit < page_size else page_size,
+            "offset": offset
+        }
+
+        while limit > 0:
+            response = self._get("_query/urls/metadata/q", params=params).json()
+            if not response:
+                break
+            for entry in response:
+                yield entry
+
+            limit -= len(response)
+            params["limit"] = min(limit, page_size)
+            params["offset"] += len(response)
+
+
+    def query_url(self, exclude=None, include=None, versioned=False, fields=None, limit=100, offset=0, page_size=100):
+        """ Queries indexd entries using URL patterns, which can be either full or partial URLs
+        Args:
+            exclude (str): A URL pattern to exclude. All URLs matching this pattern will not be included in the return
+            include (str): All entries with URL matching this pattern will be included
+            versioned (str):
+            fields:
+            limit:
+            offset:
+            page_size (int):
+        Returns:
+            Generator[Dict]
```

Comment:
Can you add some more doc string here. You can look at the doc string from the actual endpoints in [indexd](https://github.com/NCI-GDC/indexd/blob/2.8.1/indexd/urls/blueprint.py)